rclcpp/src/rclcpp/executors/executor_entities_collection.cpp

```diff
@@ -39,6 +39,30 @@ void ExecutorEntitiesCollection::clear()
   waitables.clear();
 }
 
+size_t ExecutorEntitiesCollection::remove_expired_entities()
+{
+  auto remove_entities = [](auto & collection) -> size_t {
+      size_t removed = 0;
+      for (auto it = collection.begin(); it != collection.end(); ) {
+        if (it->second.entity.expired()) {
+          ++removed;
+          it = collection.erase(it);
+        } else {
+          ++it;
+        }
```

Comment:
std::remove_if ?

`collection.erase(std::remove_if(collection.begin(), 
                              collection.end(),
                              [&removed](const auto & e) { 
                                   if(e.second.entity.expired())
                                   {
                                          removed++;
                                          return true;
                                    }
                                    return false;}), `

Reply:
I was wondering why I didn't use this, but then when I went to implement, I remembered that collection is `std::pair`, and it ends up failing compilation because 

```
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_pair.h:198:17: note: copy assignment operator is implicitly deleted because 'pair<const rcl_guard_condition_s *const, rclcpp::executors::CollectionEntry<rclcpp::GuardCondition>>' has a user-declared move constructor
      constexpr pair(pair&&) = default;         ///< Move constructor
```

Reply:
bummer :-(